src/modules/graphics/graphics.c

```diff
@@ -1660,15 +1681,16 @@ Material* lovrMaterialCreate(MaterialInfo* info) {
     for (size_t i = 0; i < state.materialBlocks.length; i++) {
       block = &state.materialBlocks.data[i];
       if (block->head != ~0u && gpu_finished(block->list[block->head].tick)) {
-        state.materialBlock = i;
+        state.materialBlock = (uint32_t)i;
         found = true;
         break;
       }
     }
 
     if (!found) {
+      lovrAssert(state.materialBlocks.length < UINT32_MAX, "Too many material blocks in system");
```

Comment:
The correct limit is `UINT16_MAX` since Material stores this number in a `uint16_t`.